src/components/POLine/POLineDetails/FieldPaymentStatus.js

```diff
@@ -13,28 +14,53 @@ const PAYMENT_STATUS = {
   pending: 'Pending',
 };
 
+const PAYMENT_STATUSES_BY_ORDER_STATUS = {
+  Pending: [
+    'pending',
+    'paymentNotRequired',
+  ],
+  Open: [
+    'partiallyPaid',
+    'paymentNotRequired',
+    'fullyPaid',
+    'cancelled',
+  ],
+};
+
 class FieldPaymentStatus extends Component {
   render() {
+    const { workflowStatus } = this.props;
+
     return (
       <Field
         component={Select}
         label={<FormattedMessage id="ui-orders.poLine.paymentStatus" />}
         name="paymentStatus"
+        disabled={!PAYMENT_STATUSES_BY_ORDER_STATUS[workflowStatus]}
       >
-        <FormattedMessage id="ui-orders.dropdown.select">
-          {(message) => <option value="">{message}</option>}
-        </FormattedMessage>
-        {Object.keys(PAYMENT_STATUS).map((key) => (
-          <FormattedMessage
-            id={`ui-orders.payment_status.${key}`}
-            key={key}
-          >
-            {(message) => <option value={PAYMENT_STATUS[key]}>{message}</option>}
-          </FormattedMessage>
-        ))}
+        {PAYMENT_STATUSES_BY_ORDER_STATUS[workflowStatus]
```

Comment:
Basically, I think that code after `.map(` you can leave as it was. Just define right option keys array in `render`

Reply:
fixed